cmd/orgs.go

```diff
@@ -215,6 +232,218 @@ func orgsRemove(ctx *cli.Context) error {
+
+	cfg, err := config.LoadConfig()
+	if err != nil {
+		return err
+	}
+
+	client := api.NewClient(cfg)
+	c := context.Background()
+
+	var org *envelope.Org
+	var orgs []envelope.Org
+
+	// Retrieve the org name supplied via the --org flag.
+	// This flag is optional. If none was supplied, then
+	// orgFlagArgument will be set to "". In this case,
+	// prompt the user to select an org.
+	orgFlagArgument := ctx.String("org")
+
+	if orgFlagArgument == "" {
+		// Retrieve list of available orgs
+		orgs, err = client.Orgs.List(c)
+		if err != nil {
+			return err
+		}
+
+		// Prompt user to select from list of existing orgs
+		idx, _, err := SelectExistingOrgPrompt(orgs)
+		if err != nil {
+			return err
+		}
+
+		org = &orgs[idx]
+
+	} else {
+		// If org flag was used, identify the org supplied.
+		org, err = client.Orgs.GetByName(c, orgFlagArgument)
+		if org == nil {
+			fmt.Println("org", orgFlagArgument, "not found.")
+			return err
+		}
+	}
+
+	// Retrieve teams, memberships and the current session concurrently
+	var teams []envelope.Team
+	var memberships []envelope.Membership
+	var session *api.Session
+	var tErr, mErr, sErr error
+
+	var getMembersTeamsSession sync.WaitGroup
+	getMembersTeamsSession.Add(3)
+
+	go func() {
+		// Retrieve list of teams in org
+		teams, tErr = client.Teams.List(c, org.ID, "", primitive.AnyTeamType)
+		if tErr != nil {
+			tErr = errs.NewExitError("Could not retrieve list of teams.")
+			getMembersTeamsSession.Done()
+			return
+		}
+
+		getMembersTeamsSession.Done()
+	}()
+
+	go func() {
+		// Retrieve list of memberships in org
+		memberships, mErr = client.Memberships.List(c, org.ID, nil, nil)
+		if mErr != nil {
+			tErr = errs.NewExitError("Could not retrieve list of memberships.")
+			getMembersTeamsSession.Done()
+			return
+		}
+		getMembersTeamsSession.Done()
+	}()
+
+	go func() {
+		// Get current session - Who am I
+		session, err = client.Session.Who(c)
+		if sErr != nil {
+			sErr = errs.NewExitError("Failed to get current session.")
+			getMembersTeamsSession.Done()
+			return
+		}
+		getMembersTeamsSession.Done()
+	}()
+
+	getMembersTeamsSession.Wait()
+	if tErr != nil || mErr != nil || sErr != nil {
+		return cli.NewMultiError(
+			tErr,
+			mErr,
+			sErr,
+		)
+	}
+
+	if len(memberships) == 0 {
+		fmt.Printf("%s has no members\n", org.Body.Name)
+		return nil
+	}
+
+	// Map team IDs to Team objects
+	teamsIdx := make(map[identity.ID]envelope.Team)
+	for _, team := range teams {
+		teamsIdx[*team.ID] = team
+	}
+
+	userTeamIdx := make(map[identity.ID][]identity.ID) // Mapping from user ID to team IDs
+	membershipUserIDs := make(map[identity.ID]bool)    // Set of unique user IDs
+
+	// Create:
+	//	- Set of unqiue user IDs in membershipUserIDs
+	// 	- Mapping from user IDs to team IDs in userTeamIdx (1:m mapping)
+	for _, membership := range memberships {
+		// Skip memberships not associated with teams within org
+		team, ok := teamsIdx[*membership.Body.TeamID]
+		if !ok {
+			continue
```

Comment:
We should probably panic here as this shouldn't be possible (a membership existing for a team that doesnt exist). If it does happen then we should get alerted!